routes/oauth.js

```diff
@@ -37,31 +37,30 @@ router.get("/init", (req, res) => {
   res.redirect(uri);
 });
 
-router.get("/redirect", (req, res) => {
+router.get("/redirect", async (req, res) => {
   if (!req.session.state) {
     // TODO: Needs better error message
     res.send("Who are you?");
     return;
   }
-  (async () => {
-    try {
-      const user = await FxAOAuth.code.getToken(req.originalUrl, { state: req.session.state });
-      const data = await popsicle.get({
-        method: "get",
-        url: FxAOAuthUtils.profileUri,
-        body: "",
-        headers: {
-          Authorization: `Bearer ${user.accessToken}`,
-        },
-      });
-      const email = JSON.parse(data.body).email;
-      await Subscribers.addUser(email);
-      res.send(`Registered ${email} for breach alerts. You may now close this window/tab.`);
-    } catch (err) {
-      console.log(err);
-      res.send(err);
-    }
-  })();
+
+  try {
+    const user = await FxAOAuth.code.getToken(req.originalUrl, { state: req.session.state });
+    const data = await popsicle.get({
+      method: "get",
+      url: FxAOAuthUtils.profileUri,
+      body: "",
+      headers: {
+        Authorization: `Bearer ${user.accessToken}`,
+      },
+    });
+    const email = JSON.parse(data.body).email;
```

Comment:
Random thought, but should we be trying to validate an email address is valid-ish before trying to create a User object (via something like [**hapijs/isemail**](https://github.com/hapijs/isemail))? Or, is that automatically done by sequelize around https://github.com/mozilla/blurts-server/pull/71/files#diff-ea1d787ec7c82cea4820cbcab09ccee5R10?

Reply:
It's automatically done by Sequelize.